src/frequenz/channels/experimental/_pipe.py

```diff
@@ -25,17 +25,34 @@ class Pipe(typing.Generic[ChannelMessageT]):
 
     Example:
         ```python
-        from frequenz.channels import Broadcast, Pipe
-
-        channel1: Broadcast[int] = Broadcast(name="channel1")
-        channel2: Broadcast[int] = Broadcast(name="channel2")
-
-        receiver_chan1 = channel1.new_receiver()
-        sender_chan2 = channel2.new_sender()
-
-        async with Pipe(channel2.new_receiver(), channel1.new_sender()):
-            await sender_chan2.send(10)
-            assert await receiver_chan1.receive() == 10
+        from contextlib import asynccontextmanager
+        from typing import AsyncIterator
+        from frequenz.channels import Broadcast, Pipe, Receiver
+
+        @asynccontextmanager
+        async def create_forwarding_pipe(
```

Comment:
Nitpick: I would try to use the pattern used by the Python library, like `closing`. Maybe call this `forwarding_channel` (as you are also returning the channel, not the pipe)?

Then:
```py
        async with forwarding_channel(
            source_channel.new_receiver()
        ) as destination_channel:
            receiver = destination_channel.new_receiver()
            await source_sender.send(10)
            assert await receiver.receive() == 10
```

Reply:
` receiver = destination_channel.new_receiver()`

You didn't closed this receiver. That is why I used asynccontextmanager :P